g-sonde/src/pages/User/Signup.tsx

```diff
@@ -0,0 +1,222 @@
+import {
+	IonButton,
+	IonCard,
+	IonCardContent,
+	IonInput,
+	IonItem,
+	IonLabel,
+	IonPage,
+	IonToast,
+} from "@ionic/react";
+
+import { IonInputCustomEvent, InputChangeEventDetail } from '@ionic/core';
+import React, { useState } from "react";
+import '../../../styles/SignUp.css'; // Assurez-vous d'utiliser le bon chemin vers votre fichier CSS
+
+import axios from "axios";
+
+
+const SignUp: React.FC = () => {
+	const [firstname, setFirstname] = useState("");
+	const [lastname, setLastname] = useState("");
+    const [email, setEmail] = useState("");
+	const [password, setPassword] = useState(""); 
+    const [confirmPassword, setConfirmPassword] = useState("");            
+	const [toastMessage, setToastMessage] = useState("");
+	const [showErrorToast, setShowErrorToast] = useState(false);
+	const [showSuccessToast, setShowSuccessToast] = useState(false);
+    const [mailValide, setMailValide] = useState(false);
+    const [mdpValide, setMdpValide] = useState(false);
+    const [checkMail, setCheckMail] = useState(false);
+    const [backgroundColor, setBackgroundColor] = useState("");
+    
```

Comment:
Je pense que ici tu pourrais plutôt utiliser un objet pour contenir tes states plutot que d'allouer un state par valeur

cela pourrait t'aider : https://legacy.reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables

Reply:
Je lirai le lien et amaliorai mon code. Merci